src/types.rs

```diff
@@ -111,3 +111,58 @@ impl TryFrom<u8> for Channels {
         }
     }
 }
+
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum RawChannels {
+    Rgb,
+    Bgr,
+    Rgba,
+    Argb,
+    Rgbx,
+    Xrgb,
+    Bgra,
+    Abgr,
+    Bgrx,
+    Xbgr,
```

Comment:
A docstring for each variant? Might not be clear to all e.g. what xbgr is and how it is handled here.